drivers/sensor/bme680/bme680.c

```diff
@@ -0,0 +1,427 @@
+/* bme680.c - Driver for Bosch Sensortec's BME680 temperature, pressure,
+ * humidity and gas sensor
+ *
+ * https://www.bosch-sensortec.com/bst/products/all_products/bme680
+ */
+
+/*
+ * Copyright (c) 2016, 2017 Intel Corporation
+ * Copyright (c) 2017 IpTronix S.r.l.
+ * Copyright (c) 2018 Bosch Sensortec GmbH
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "bme680.h"
+#include <gpio.h>
+#include <i2c.h>
+#include <init.h>
+#include <kernel.h>
+#include <misc/byteorder.h>
+#include <misc/__assert.h>
+#include <sensor.h>
+
+static int bme680_reg_read(struct bme680_data *data, u8_t start, u8_t *buf,
+			   int size)
+{
+	return i2c_burst_read(data->i2c_master, data->i2c_slave_addr, start,
+			      buf, size);
+	return 0;
+}
+
+static int bme680_reg_write(struct bme680_data *data, u8_t reg, u8_t val)
```

Comment:
These i2c wrapper functions are overkill.

Reply:
It would be easier when we update it with support for the SPI interface. The SPI interface is operated over 2 memory pages and the switching of the memory pages will be handled within this function. If the I2C functions were directly implemented, adding SPI support would have more rework at the end.